pkg/node/fsm/ng_state.go

```diff
@@ -205,6 +207,35 @@ func (a *NGState) Block(peer peer.Peer, block *proto.Block) (State, Async, error
 	return newNGState(a.baseInfo), nil, nil
 }
 
+func (a *NGState) BlockEndorsement(blockEndorsement *proto.EndorseBlock) (State, Async, error) {
+	endorsedBlockID := blockEndorsement.EndorsedBlockID
+	endorsedMicroBlock, found := a.baseInfo.MicroBlockCache.GetBlock(endorsedBlockID)
+	if !found || endorsedMicroBlock == nil {
+		return a, nil, a.Errorf(fmt.Errorf(
+			"failed to find the microblock that was endorsed (block ID: %s)",
+			endorsedBlockID.String(),
+		))
+	}
+	top := a.baseInfo.storage.TopBlock()
+	if top.BlockID() != endorsedMicroBlock.Reference { // Microblock's endorsement doesn't refer to last block
```

Comment:
As I know, endorsements are sent for parent blocks. It means that `top.BlockID` never equal `endorsedMicroBlock.Reference`, so the condition is always true.